WordPressKit/WordPressKit/AccountServiceRemoteREST.m

```diff
@@ -193,13 +193,47 @@ - (void)requestWPComAuthLinkForEmail:(NSString *)email
                             clientID:(NSString *)clientID
                         clientSecret:(NSString *)clientSecret
                          wpcomScheme:(NSString *)scheme
-                             success:(void (^)(void))success failure:(void (^)(NSError *error))failure
+                             success:(void (^)(void))success
+                             failure:(void (^)(NSError *error))failure
 {
-    NSAssert([email length] > 0, @"Needs an email address.");
+    [self requestWPComLinkForEmail:email
+                      endpointPath:@"auth/send-login-email"
+                          clientID:clientID
+                      clientSecret:clientSecret
+                       wpcomScheme:scheme
+                           success:success
+                           failure:failure];
+}
 
-    NSString *path = [self pathForEndpoint:@"auth/send-login-email"
-                                     withVersion:ServiceRemoteWordPressComRESTApiVersion_1_1];
+- (void)requestWPComSignupLinkForEmail:(NSString *)email
+                              clientID:(NSString *)clientID
+                          clientSecret:(NSString *)clientSecret
+                           wpcomScheme:(NSString *)scheme
+                               success:(void (^)(void))success
+                               failure:(void (^)(NSError *error))failure
+{
+    [self requestWPComLinkForEmail:email
+                      endpointPath:@"auth/send-signup-email"
+                          clientID:clientID
+                      clientSecret:clientSecret
+                       wpcomScheme:scheme
+                           success:success
+                           failure:failure];
+}
 
+- (void)requestWPComLinkForEmail:(NSString *)email
```

Comment:
imo `requestWPComLinkForEmail` is a big ambiguous, as it's not just any link, it's a magic link. How about `requestWPComMagicLinkForEmail`?

Reply:
Sounds good; I can rename it.